iggy/build.rs

```diff
@@ -0,0 +1,267 @@
+                    entry.signature_wildcard_pattern,
+                    entry.snake_case_name,
+                    indent = crate::get_spaces(num_spaces),
+                )
+            }
+            ConversionType::AsCode => {
+                format!(
+                    "{indent}Error::{}{} => {},",
+                    crate::snake_to_pascal_case(&entry.snake_case_name),
+                    entry.signature_wildcard_pattern,
+                    entry.code,
+                    indent = crate::get_spaces(num_spaces),
+                )
+            }
+            ConversionType::FromCodeAsString => {
+                format!(
+                    "{indent}{} => \"{}\",",
+                    entry.code,
+                    entry.snake_case_name,
+                    indent = crate::get_spaces(num_spaces),
+                )
+            }
+        }
+    }
+
+    fn create_arms(
+        &self,
+        preprocessed_error_codes: &Vec<PreprocessedErrorRepositoryEntry>,
+        default_value: Option<String>,
+    ) -> Vec<String> {
+        let mut result: Vec<String> = vec![];
+
+        for (idx, preprocessed_error_code) in preprocessed_error_codes.iter().enumerate() {
+            let num_spaces = if idx == 0 { 0 } else { 12 };
+
+            let next = self.to_match_arm(&preprocessed_error_code, num_spaces);
+            result.push(next);
+        }
+
+        if let Some(dv) = default_value {
+            let indent = get_spaces(12);
+            result.push(format!("{}_ => \"{}\"", indent, dv));
+        }
+
+        result
+    }
+}
+
+struct MatchConversionFunction {
+    name: String,
+    parameter_list: String,
+    return_type: String,
+    match_arms: Vec<String>,
+    match_on: String,
+}
+
+impl MatchConversionFunction {
+    fn to_code_string(&self) -> String {
+        let lines = vec![
+            format!(
+                "{indent}{signature} {{",
+                indent = get_spaces(4),
+                signature = format!(
+                    "pub fn {name}({parameter_list}) -> {return_type}",
+                    name = self.name,
+                    parameter_list = self.parameter_list,
+                    return_type = self.return_type
+                )
+            ),
+            format!(
+                "{indent}match {match_on} {{",
+                indent = get_spaces(8),
+                match_on = self.match_on
+            ),
+            format!(
+                "{indent}{as_code_arms}",
+                as_code_arms = self.match_arms.join("\n"),
+                indent = get_spaces(12)
+            ),
+            format!("{indent}}}", indent = get_spaces(8)),
+            format!("{indent}}}\n", indent = get_spaces(4)),
+        ];
+
+        lines.join("\n")
+    }
+}
+
+const GENERATED_ERRORS_PATH: &'static str = "./src/errors/generated_code/errors.rs";
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let errors_db: SledDb = get_or_create().expect("Could not load errors database");
```

Comment:
stick to one way of handling errors: either use expect() everywhere, or return with syntax sugar `?` like you did at line 206. I prefer second option.

Reply:
Updated

Reply:
Also added the rustfmt step